modules/boards/app/controllers/boards/boards_controller.rb

```diff
@@ -37,6 +37,8 @@ def overview
     def new; end
 
     def create
+      return render_403 if restricted_board_type?
```

Comment:
I'd prefer to see this as a before action to prevent the extra return, but that's just my preference

Reply:
Addressed in https://github.com/opf/openproject/pull/13259/commits/f82d2e3588cbcb599daf85bccfec65b317d90875